### PR TITLE
Me county fips code

### PR DESCRIPTION
**Pull request by saipraveen18**

### CODE REVIEW

File: app/models/address.rb [hunk 1]
```diff
@@ -44,7 +45,7 @@ def clean_fields
 
   def match(another_address)
     return(false) if another_address.nil?
-    attrs_to_match = [:address_type, :address_1, :address_2, :city, :state, :zip]
+    attrs_to_match = [:address_type, :address_1, :address_2, :city, :state, :zip, :location_county_code]
```
Comment (TreyE) [inline app/models/address.rb:48]: Shouldn't this also include county name in the comparison?

File: app/models/remote_resources/address.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ class Address
     element :address_line_2, String, tag: "address_line_2"
     element :location_city_name, String, tag: "location_city_name"
     element :location_state_code, String, tag: "location_state_code"
+    element :location_county_code, String, tag: "location_county_code"
```
Comment (TreyE) [inline app/models/remote_resources/address.rb:14]: Shouldn't we also check for county name here as well if we have it?

File: app/models/parsers/edi/etf/responsible_party_parser.rb [hunk 3]
```diff
@@ -97,12 +100,14 @@ def parse_address
             city = @rp_loop["N4"][1]
             state = @rp_loop["N4"][2]
             zip = @rp_loop["N4"][3]
+            county_code = @loop["L2100A"]["N4"][4].blank? ? nil : @loop["L2100A"]["N4"][4]
```
Comment (TreyE) [inline app/models/parsers/edi/etf/responsible_party_parser.rb:103]: Number of issues here:
1. FIPS codes aren't present in the N4 for responsible parties.
2. You won't find responsible party info in the 2100A loop.  It lives in G or H or something like that.

File: app/models/parsers/edi/etf/person_loop.rb [hunk 4]
```diff
@@ -111,6 +111,11 @@ def zip
           @loop["L2100A"]["N4"][3]
         end
 
+        def county_code
+          result = @loop["L2100A"]["N4"][4]
+          ((result.blank?) ? nil : result)
+        end
```
Comment (TreyE) [inline app/models/parsers/edi/etf/person_loop.rb:117]: County code, if it exists, lives in the N406 loop (index `6` in the N4 array) and this is if and only if the N405 (element `5` in the array) is present with a value of 'CY'.

File: spec/models/parsers/edi/etf/person_parser_spec.rb [hunk 5]
```diff
@@ -8,7 +8,7 @@
     {
       'L2100A' => {
         "N3" => ['', street1, street2, ''],
-        'N4' => ['', city, state, zip]
+        'N4' => ['', city, state, zip, county_code]
```
Comment (TreyE) [inline spec/models/parsers/edi/etf/person_parser_spec.rb:11]: Same deal with N4 structure as mentioned above.  N404 is actually the **country** code - which we never provide.

**Review by TreyE — CHANGES_REQUESTED** [review]
Please see my notes.

File: app/models/address.rb (hunk 1 above)
Comment (saipraveen18) [reply app/models/address.rb:48]: Currently, we are not sending County name from upstream (EA)

File: app/models/remote_resources/address.rb (hunk 2 above)
Comment (saipraveen18) [reply app/models/remote_resources/address.rb:14]: Currently, we are not sending County name from upstream (EA)

File: app/models/parsers/edi/etf/person_loop.rb (hunk 4 above)
Comment (saipraveen18) [reply app/models/parsers/edi/etf/person_loop.rb:118]: Fixed

File: app/models/address.rb (hunk 1 above)
Comment (TreyE) [reply app/models/address.rb:48]: Correct

**Review by TreyE — APPROVED** [review]
Looks correct.

**Pull request merged**